cogs/pot_tres_confine.py

```diff
@@ -0,0 +1,146 @@
+        await category.delete()
+
+    @commands.command()
+    @has_orga_soiree_role()
+    async def repartition(self, ctx: commands.context.Context, max_by_channel: int = 8):
+        """
+        Distributes the participants placed in the general voice channel to random lobbies
+        :param ctx: context object
+        :param max_by_channel: number of maximum players placed in a channel
+        """
+        category = MyUtils(ctx.guild).getPotTresConfineCategory()
+        general_channel_id = discord.utils.get(category.voice_channels, name="General").id
+        member_ids = list(self.bot.get_channel(general_channel_id).voice_states.keys())
+        nb_users = len(member_ids)
+
+        # Create the right number of lobbies
+        for i in range(int((nb_users / max_by_channel)) + 1):
+            await ctx.guild.create_voice_channel(f"lobby-{i + 1}", category=category)
+
+        voice_channels = category.voice_channels
+        shuffle(member_ids)  # Randomize the ids
+
+        # Move the participant to the right channel
+        for index, _ in enumerate(member_ids):
+            member = await ctx.guild.fetch_member(member_ids[index])
+            await member.move_to(voice_channels[index // max_by_channel + 1])
+
+    @commands.command()
+    @has_orga_soiree_role()
+    async def rassemblement(self, ctx: commands.context.Context):
+        """
+        Move every participant in the lobbies to the general channel
+        :param ctx: context object
+        """
+        category = MyUtils(ctx.guild).getPotTresConfineCategory()
+        general_channel_id = discord.utils.get(category.voice_channels, name="General").id
+        voice_channel = category.voice_channels
+        voice_channel.pop(0)
+
+        for i in range(len(voice_channel)):
+            members = voice_channel[i].voice_states
+            for member_id in members.keys():
+                member = await ctx.guild.fetch_member(member_id)
+                await member.move_to(self.bot.get_channel(general_channel_id))
+            await voice_channel[i].delete()
+
+    @commands.command()
+    @has_orga_soiree_role()
+    async def mute_all(self, ctx: commands.context.Context):
+        """
+        Mute every participant in the voice channel where the author is, except the orga_soiree
+        :param ctx: context object
+        :param mute_type: str
+        """
+        if not (ctx.author.voice and ctx.author.voice.channel):
+            await ctx.send("You are not connected to a voice channel")
+            return
+        channel = ctx.author.voice.channel
+
+        members = channel.voice_states
+        for member_id in members.keys():
+            member = await ctx.guild.fetch_member(member_id)
+            if MyUtils(ctx.guild).getOrgaSoireeRole() not in member.roles:
+                await member.edit(mute=True)
+
+    @commands.command()
+    @has_orga_soiree_role()
+    async def demute_all(self, ctx: commands.context.Context):
```

Comment:
Ca ne marche pas bien ca chez moi ^^

Reply:
Il te manque un not dans ton if pour que ca fonctionne ;)

Reply:
en fait ça mute ceux qui n'ont pas le rôle orga soirée, donc faut être 2 pour le tester

Reply:
ca fait bien ca :) 
mais pour demute ca ne le fait pas

Reply:
ligne 137 `if not (ctx.author.voice and ctx.author.voice.channel):`

Reply:
rha mb on a refait la condition au dernier moment

Reply:
yes, quand on a test avec le `not` ca mute tout le monde sauf les orga